docs/reference/mapping/types/search-as-you-type.asciidoc

```diff
@@ -0,0 +1,184 @@
+[[search-as-you-type]]
+=== Search as you type datatype
+
+experimental[]
+
+The `search_as_you_type` field type is a text-like field that is optimized to
+provide out-of-the-box support for queries that serve an as-you-type completion
+use case.
+
+This field type creates a series of subfields that are analyzed to index terms
+that can be efficiently matched by a query that partially matches the entire
+indexed text value. Both prefix completion (i.e matching terms starting at the
+beginning of the input) and infix completion (i.e. matching terms at any
+position within the input) are supported.
+
+When adding a field of this type to a mapping
+
+[source,js]
+--------------------------------------------------
+PUT my_index
+{
+  "mappings": {
+    "properties": {
+      "my_field": {
+        "type": "search_as_you_type"
+      }
+    }
+  }
+}
+--------------------------------------------------
+// CONSOLE
+
+This creates the following fields
+
+[horizontal]
+
+`my_field`::
+
+    Analyzed as configured in the mapping. If an analyzer is not configured,
+    the default analyzer for the index is used
+
+`my_field._2gram`::
+
+    Wraps the analyzer of `my_field` with a shingle token filter of shingle
+    size 2
+
+`my_field._index_prefix`::
+
+    Wraps the analyzer of `my_field._2gram` with an edge ngram token filter
```

Comment:
I omitted an explanation of the filter that adds position increments at the end to create the trailing shingles - I thought it might be confusing here and better illustrated in example queries

Reply:
++, I expect this field to be used only for expert queries. It should also not be very useful to document it once the `match_prefix` query is added. Maybe we should omit it but add a section regarding how prefix queries are optimized ?